s3/src/main/java/ch/cyberduck/core/sts/AssumeRoleWithWebIdentitySTSCredentialsConfigurator.java

```diff
@@ -16,45 +16,62 @@
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AnonymousAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
 import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
 import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
 import com.amazonaws.services.securitytoken.model.AssumeRoleWithWebIdentityRequest;
 import com.amazonaws.services.securitytoken.model.AssumeRoleWithWebIdentityResult;
 
-public class NonAwsSTSCredentialsConfigurator extends STSCredentialsConfigurator {
-    private static final Logger log = LogManager.getLogger(NonAwsSTSCredentialsConfigurator.class);
+public class AssumeRoleWithWebIdentitySTSCredentialsConfigurator extends STSCredentialsConfigurator {
+    private static final Logger log = LogManager.getLogger(AssumeRoleWithWebIdentitySTSCredentialsConfigurator.class);
 
 
-    public NonAwsSTSCredentialsConfigurator(final X509TrustManager trust, final X509KeyManager key, final PasswordCallback prompt) {
-       super(trust, key, prompt);
+    public AssumeRoleWithWebIdentitySTSCredentialsConfigurator(final X509TrustManager trust, final X509KeyManager key, final PasswordCallback prompt) {
+        super(trust, key, prompt);
     }
 
     @Override
     public Credentials configure(final Host host) throws LoginFailureException, LoginCanceledException {
         final Credentials credentials = new Credentials(host.getCredentials());
-        final ClientConfiguration configuration = new CustomClientConfiguration(host,
-                new ThreadLocalHostnameDelegatingTrustManager(trust, host.getHostname()), key);
-
 
+        // STS API is open, no authorization required
         final AWSSecurityTokenService service = AWSSecurityTokenServiceClientBuilder
                 .standard()
-                .withClientConfiguration(configuration)
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(String.format("%s://%s:%s", host.getProtocol().getScheme(), host.getHostname(),
+                        host.getPort()), null))
+                .withCredentials(new AWSCredentialsProvider() {
+                    @Override
+                    public AWSCredentials getCredentials() {
+                        //         // https://www.demo2s.com/java/amazon-aws-assumerolewithwebidentityrequest-tutorial-with-examples.html
+                        return new AnonymousAWSCredentials();
+                    }
+
+                    @Override
+                    public void refresh() {
+
+                    }
+                })
                 .build();
 
+
         AssumeRoleWithWebIdentityRequest webIdReq = new AssumeRoleWithWebIdentityRequest()
-                .withWebIdentityToken("token")
-                .withPolicy("policy");
+                .withWebIdentityToken(credentials.getOauth().getAccessToken());
+        // TODO check with DK: make configurable in profile/bookmark?
```

Comment:
Should be configurable in connection profile and fallback to AWS CLI configuration if any.

Reply:
Outdated -> resolve.